collect_app/src/main/java/org/odk/collect/android/preferences/AdminPreferencesFragment.java

```diff
@@ -0,0 +1,123 @@
+package org.odk.collect.android.preferences;
+
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.preference.ListPreference;
+import android.preference.Preference;
+import android.preference.PreferenceFragment;
+import android.preference.PreferenceManager;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.WindowManager;
+import android.widget.EditText;
+
+import org.odk.collect.android.R;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.utilities.ToastUtils;
+
+import static android.content.Context.MODE_PRIVATE;
+import static android.content.Context.MODE_WORLD_READABLE;
+import static org.odk.collect.android.preferences.AdminKeys.KEY_ADMIN_PW;
+import static org.odk.collect.android.preferences.AdminKeys.KEY_CHANGE_ADMIN_PASSWORD;
+import static org.odk.collect.android.preferences.AdminKeys.KEY_FORM_PROCESSING_LOGIC;
+
+
+public class AdminPreferencesFragment extends PreferenceFragment implements Preference.OnPreferenceClickListener, Preference.OnPreferenceChangeListener {
+
+    public static String ADMIN_PREFERENCES = "admin_prefs";
+
+    @Override
+    public void onCreate(final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        PreferenceManager prefMgr = getPreferenceManager();
+        prefMgr.setSharedPreferencesName(ADMIN_PREFERENCES);
+        prefMgr.setSharedPreferencesMode(MODE_WORLD_READABLE);
+
+        addPreferencesFromResource(R.xml.admin_preferences);
+
+        ListPreference mFormProcessingLogicPreference = (ListPreference) findPreference(KEY_FORM_PROCESSING_LOGIC);
+        mFormProcessingLogicPreference.setSummary(mFormProcessingLogicPreference.getEntry());
+        mFormProcessingLogicPreference.setOnPreferenceChangeListener(this);
+
+        Preference mChangeAdminPwPreference = findPreference(KEY_CHANGE_ADMIN_PASSWORD);
+        mChangeAdminPwPreference.setOnPreferenceClickListener(this);
+    }
+
+    @Override
+    public boolean onPreferenceClick(Preference preference) {
+
+        if (preference.getKey().equals(KEY_CHANGE_ADMIN_PASSWORD)) {
+            AlertDialog.Builder b = new AlertDialog.Builder(getActivity());
```

Comment:
Yes, good point by @dcbriccetti. We should be picky style-wise about new code but it's ok to leave old things alone to minimize invasive changes.